libsawtooth/src/client/rest.rs

```diff
@@ -46,34 +46,11 @@ impl SawtoothClient for RestApiSawtoothClient {
     /// Get the batch with the given batch_id from the current blockchain
     fn get_batch(&self, batch_id: String) -> Result<Option<ClientBatch>, SawtoothClientError> {
         let url = format!("{}/batches/{}", &self.url, &batch_id);
-        let request = Client::new().get(&url);
-        let response = request
-            .send()
-            .map_err(|err| SawtoothClientError::new_with_source("request failed", err.into()))?;
 
-        if response.status().is_success() {
-            let batch: Single<Batch> = response.json().map_err(|err| {
-                SawtoothClientError::new_with_source(
-                    "failed to deserialize response body",
-                    err.into(),
-                )
-            })?;
-            let clientbatch: ClientBatch = batch.data.into();
-            Ok(Some(clientbatch))
-        } else if response.status().as_u16() == 404 {
-            Ok(None)
+        if let Some(batch) = get::<Batch>(&url).unwrap() {
```

Comment:
Shouldn't use `unwrap` here. This whole `if let .. else ...` can just be simplified to something like `get::<Batch>(&url).map(|opt| opt.map(ClientBatch::from))`

Reply:
Agreed.  It's a good rule here to use no `unwrap` calls.  There are a few exceptions, but this doesn't fit that category.  What you should do, instead of the `unwrap` is either @ltseeley's suggestion or just replace `unwrap` with `?`, so the error result is propagated out of the function.

Reply:
fixed